### PR TITLE
Sanitize `--repo-branch` input in `inv pipeline.trigger`

### PR DESCRIPTION
**Pull request by KSerrania**
### What does this PR do?

Sanitizes `--repo-branch` input in `inv pipeline.trigger`.

### Motivation

Avoid uploading artifacts to wrong branches.

### CODE REVIEW

File: tasks/pipeline.py [hunk 1]
```diff
@@ -18,6 +18,8 @@
 
 # Tasks to trigger pipelines
 
+ALLOWED_REPO_BRANCHES = ["stable", "beta", "nightly", "none"]
```
Comment (mx-psi) [inline tasks/pipeline.py:21]:
```suggestion
ALLOWED_REPO_BRANCHES = {"stable", "beta", "nightly", "none"}
```

File: tasks/pipeline.py [hunk 2]
```diff
@@ -37,6 +39,15 @@ def trigger(_, git_ref="master", release_version_6="nightly", release_version_7=
     gitlab = Gitlab()
     gitlab.test_project_found(project_name)
 
+    # Check that the target repo branch is valid
+    if repo_branch not in ALLOWED_REPO_BRANCHES:
+        print(
+            "--repo-branch argument '{}' is not in the list of allowed repsoitory branches: {}".format(
```
Comment (mx-psi) [inline tasks/pipeline.py:45]:
```suggestion
            "--repo-branch argument '{}' is not in the list of allowed repository branches: {}".format(
```

**Review by albertvaka — APPROVED** [review]
🍰

File: tasks/pipeline.py [hunk 3]
```diff
@@ -18,6 +18,8 @@
 
 # Tasks to trigger pipelines
 
+ALLOWED_REPO_BRANCHES = {"stable", "beta", "nightly", "none"}
```
Comment (albertvaka) [inline tasks/pipeline.py:21]: Does `none` have a special meaning? Where is it treated?

**Review by mx-psi — APPROVED** [review]

File: tasks/pipeline.py (hunk 3 above)
Comment (KSerrania) [reply tasks/pipeline.py:21]: Historically, setting the repo branch to `none` was the way to run a deploy pipeline without actually sending the artifacts anywhere. It's used in the `check_if_build_only` job:
https://github.com/DataDog/datadog-agent/blob/3eb5ceb4f7bba02da244b412bd1ac34e9367c967/.gitlab/check_deploy.yml#L49-L58

However, I'm not sure if that still has a purpose: maybe we can remove that option (and the associated job).

**Pull request merged**